tests/avocado/test_numa_hotplug.py

```diff
@@ -20,6 +20,15 @@ class TestNumaHotplug(test.QemuTest):
     """
 
     def setUp(self):
+        self.image = vmimage.get('Fedora')
+
+        virt_customize = utils_path.find_command('virt-customize')
+        process.run('%s --add %s --password %s:password:%s --uninstall %s' %
+                    (virt_customize, self.image, 'root', '1234', 'cloud-init'))
+
+        self.vm.add_image(self.image, 'root', '1234', snapshot=False)
+        self.vm.add_machine(machine_accel='kvm')
```

Comment:
well this should be already added by `params`. I'm wondering how to properly treat things like that (test requires something, that is by default set by params):

1. First process the defaults from `params` during `launch`, which would allow to combine the defaults set by the test with the params
2. Skip tests that use clashing params (in this example it'd be `if params.get("machine_accel") != "kvm": self.skip(...)`)

The (1) is easier to use while the (2) works better when executing test matrices. As we are focused on developers I'd go with (1) and instead of directly creating `args` I'd only store the params and on `launch` I'd take a look at what was already requested and add the remaining bits.

Also looking at this I believe we need a concept of defaults as for example `arm` requires `-machine` otherwise it refuses to do anything, while x86_64 simply uses the default.